### PR TITLE
fix!: change the instrumentation config to circumvent viper bug

### LINKED ISSUE #1141: The trace client fails to restrict which tables are traced
The trace client is supposed to have the ability to select what is traced. This is currently not case, where everything gets traced. This becomes problematic in testground tests, where we need to collect specific data from all nodes, but must avoid overloading an influx instance with all trace data from 100+ nodes.

### PR DESCRIPTION
**Pull request by evan-forbes**
## Description

There's an viper property or potential bug that stops us from overriding the default values for slices of strings. The "solution" atm is to use a string and parse it later like animals later. Unfortunately I think this is breaking since we're changing what a config is doing so tbh I'm not sure if we can ever include this in v1. If that's the case then we might want to consider not merging this to v0.34.x-celestia and only to main.

closes #1141

### CODE REVIEW

**Review by rootulp — APPROVED** [review]
> Unfortunately I think this is breaking since we're changing what a config is doing so tbh I'm not sure if we can ever include this in v1. If that's the case then we might want to consider not merging this to v0.34.x-celestia and only to main.

I'm in favor of merging this and cutting a new minor release (e.g. v1.30.0-tm-v0.34.29). I realize that technically breaks SemVer but I think the pro outweights the con.

Pro: feature parity between the branches we maintainer
Con: breaking SemVer for releases on this repo.

I'm thinking this because we expect the # of consumers of celestia-core to be very few (ideally just celestia-app) and since we're responsible for bumping this dep in celestia-app, we know to fix the config when we bump this dep in celestia-app.

File: config/toml.go [hunk 1]
```diff
@@ -558,8 +558,9 @@ influx_org = "{{ .Instrumentation.InfluxOrg }}"
 influx_batch_size = {{ .Instrumentation.InfluxBatchSize }}
 
 # The list of tables that are updated when tracing. All available tables and
-# their schema can be found in the pkg/trace/schema package.
-influx_tables = [{{ range .Instrumentation.InfluxTables }}{{ printf "%q, " . }}{{end}}]
+# their schema can be found in the pkg/trace/schema package. It is represented as a
+// comma separate string. For example: "consensus_round_state,mempool_tx".
```
Comment (rootulp) [inline config/toml.go:562]:
```suggestion
# comma separate string. For example: "consensus_round_state,mempool_tx".
```

File: config/toml.go [hunk 2]
```diff
@@ -572,8 +573,9 @@ pyroscope_trace = {{ .Instrumentation.PyroscopeTrace }}
 # pyroscope_profile_types is a list of profile types to be traced with
 # pyroscope. Available profile types are: cpu, alloc_objects, alloc_space,
 # inuse_objects, inuse_space, goroutines, mutex_count, mutex_duration,
-# block_count, block_duration.
-pyroscope_profile_types = [{{ range .Instrumentation.PyroscopeProfileTypes }}{{ printf "%q, " . }}{{end}}]
+# block_count, block_duration. It is represented as a comma separate
+// string. For example: "goroutines,alloc_objects".
```
Comment (rootulp) [inline config/toml.go:577]:
```suggestion
# string. For example: "goroutines,alloc_objects".
```

**Review by rootulp — APPROVED** [review]

**Review by staheri14 — APPROVED** [review]

**Comment by evan-forbes** [comment]
:hand_over_mouth: whoops thanks for the comment fix @rootulp

**Pull request merged**